extensions/vertx/runtime/src/main/java/io/quarkus/vertx/runtime/VertxCurrentContextFactory.java

```diff
@@ -0,0 +1,57 @@
+package io.quarkus.vertx.runtime;
+
+import java.lang.annotation.Annotation;
+
+import io.quarkus.arc.CurrentContext;
+import io.quarkus.arc.CurrentContextFactory;
+import io.quarkus.arc.InjectableContext;
+import io.quarkus.arc.InjectableContext.ContextState;
+import io.quarkus.vertx.core.runtime.context.VertxContextSafetyToggle;
+import io.smallrye.common.vertx.VertxContext;
+import io.vertx.core.Context;
+import io.vertx.core.Vertx;
+
+public class VertxCurrentContextFactory implements CurrentContextFactory {
+
+    @Override
+    public <T extends InjectableContext.ContextState> CurrentContext<T> create(Class<? extends Annotation> scope) {
+        return new VertxCurrentContext<>();
+    }
+
+    private static final class VertxCurrentContext<T extends ContextState> implements CurrentContext<T> {
+
+        private final ThreadLocal<T> fallback = new ThreadLocal<>();
```

Comment:
Although a fallback, probably worth using FastThreadLocal here

Reply:
Yep, and will fall-back to ThreadLocal if it's not - meaning that reactive stacks (running on I/O loops) will benefit by it

Reply:
The blocking thread pool (using JBoss Executor) is using it too IIRC

Reply:
yes, here we can use a FastThreadLocal.

I will have to check with @anavarr how we deal with all these fast thread local in the loom (as we remove them, if I'm not mistaken).

Reply:
Ok, I'll change it then...